.github/workflows/ci.yml

```diff
@@ -0,0 +1,106 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  primary_code_validation_and_tests:
+    name: Primary code validation and tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 12
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+
+      # This also runs a build as part of the postinstall bootstrap
+      - name: install and build
+        run: |
+          yarn --ignore-engines --frozen-lockfile
+          yarn check-clean-workspace-after-install
+
+      # Note that this command *also* typechecks tests/tools,
+      # whereas the build only checks src files
+      - name: Typecheck all packages
+        run: yarn typecheck
+
+      - name: Check code formatting
+        run: yarn format-check
+
+      - name: Run linting
+        run: yarn lint
+
+      - name: Validate spelling
+        run: yarn check:spelling
+
+      - name: Run unit tests
+        run: yarn test
+        env:
+          CI: true
+
+      - name: Run integrations tests
+        run: yarn integration-tests
+        env:
+          CI: true
+
+      - name: Publish code coverage report
+        uses: codecov/codecov-action@v1
+        with:
+          yml: ./codecov.yml
+          token: ${{ secrets.CODECOV_TOKEN }}
+          flags: unittest
+          name: codecov
```

Comment:
Question - did we want to split all of this up to give us more granular status checks?
I know that a bit of a pain point for people is just seeing "primary validation" failed, and then having to dig into the logs to see which step failed.

I.e. one job for each of:
- type check
- format check
- lint
- spelling
- test + integration (+ coverage publish) test on latest node

Reply:
you actually don't have to as actions are showing it in overview screen